src/ngx_http_lua_common.h

```diff
@@ -176,6 +178,9 @@ struct ngx_http_lua_main_conf_s {
 
     ngx_array_t         *preload_hooks; /* of ngx_http_lua_preload_hook_t */
 
+    ngx_array_t         *delay_list; /* of void* */
+    unsigned             delay_list_inited;
```

Comment:
@agentzh using an array here because that I just want to make it like `required_shm`, we can run it again and again in a same third part module, and it is ported from `ngx_shared_memory_add`.
Maybe we just use a counter here and force mandatory third part module just run once in their own init phase?